### PR TITLE
Consolidate Categories and ReportCategories classes

### PR DESCRIPTION
**Pull request by Pyifan**
There 2 classes are overlapping, remove Categories.
Also change TESTCASE = 'testcase' --> CASE = 'case' to be consistent with SUITE = 'suite'

### CODE REVIEW

File: examples/Test Output/Exporters/JSON/json_to_pdf.py [hunk 1]
```diff
@@ -5,7 +5,7 @@
 import argparse
 import json
 
-from testplan.report.testing import TestReport
+from testplan.report import TestReport
```
Comment (ryan-collingham) [inline examples/Test Output/Exporters/JSON/json_to_pdf.py]: If this import was missing before... how did it ever work?

File: testplan/common/utils/logger.py [hunk 2]
```diff
@@ -15,7 +15,7 @@
 import logging
 
 from testplan.common.utils.strings import Color
-from testplan.report.testing import Status
+from testplan.report import Status
```
Comment (ryan-collingham) [inline testplan/common/utils/logger.py]: Again, how was this not broken before...?

File: testplan/report/testing/base.py [hunk 3]
```diff
@@ -136,19 +136,19 @@ class ReportCategories(object):
     """
 
     TESTPLAN = "testplan"
-    TESTGROUP = "testgroup"
+    TESTGROUP = "testgroup"     # test group of unspecific type?
     MULTITEST = "multitest"
+    SUITE = "suite"
+    CASE = "case"
```
Comment (ryan-collingham) [inline testplan/report/testing/base.py]: This will break the UI since it expects testcases to have a category of "testcase". Please can you change back the value to "testcase"?

File: testplan/runnable/interactive/base.py [hunk 4]
```diff
@@ -662,7 +662,7 @@ def _initial_report(self):
             for suite in test.suites:
                 suite_name = suite.__class__.__name__
                 suite_report = testplan.report.TestGroupReport(
-                    name=suite_name, uid=suite_name, category="suite"
+                    name=suite_name, uid=suite_name, category='suite'
```
Comment (ryan-collingham) [inline testplan/runnable/interactive/base.py]: Why have you made this change? Double quotes are the standard used by `black`.

File: testplan/testing/multitest/base.py [hunk 5]
```diff
@@ -249,7 +242,7 @@ def pre_post_step_report(self):
         if self._pre_post_step_report is None:
             self._pre_post_step_report = TestGroupReport(
                 name='Pre/Post Step Checks',
-                category=Categories.SUITE,
+                category=ReportCategories.SUITE,
```
Comment (ryan-collingham) [inline testplan/testing/multitest/base.py]: Nice, better to use the ReportCategories values here for sure.

File: tests/functional/testplan/runnable/interactive/test_api.py [hunk 6]
```diff
@@ -160,7 +160,7 @@ def plan():
         "/report/tests/ExampleMTest/suites/ExampleSuite/testcases",
         [
             {
-                "category": "testcase",
+                "category": "case",
```
Comment (ryan-collingham) [inline tests/functional/testplan/runnable/interactive/test_api.py]: This will be a problem, the UI won't handle the value of "case".

File: tests/functional/testplan/testing/fixtures/base/passing/report.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-from testplan.report.testing import TestReport, TestGroupReport
+from testplan.report import TestReport, TestGroupReport
```
Comment (ryan-collingham) [inline tests/functional/testplan/testing/fixtures/base/passing/report.py]: A lot of unrelated changes being made... it's fine for now but in future could you split out such cosmetic changes into a separate PR?

File: testplan/report/testing/base.py (hunk 3 above)
Comment (Pyifan) [reply testplan/report/testing/base.py]: Yea, i was debating whether I should change suite to testsuite or change testcase to case. I recon the latter one will have smaller impact as the CASE type is lately added by you recently. I prefer to keep the terminologies consistent, either testsuite/testcase or suite/case, but definitely not testsuite/case. We can do a global replace in the UI code from testcase -> case.

Comment (ryan-collingham) [reply testplan/report/testing/base.py]: as discussed, let's go with testsuite/testcase and be consistent everywhere please.

File: testplan/web_ui/testing/src/Common/defaults.js [hunk 8]
```diff
@@ -45,7 +45,7 @@ const CATEGORIES = {
   'boost-test-suite': 'suite',
   'gtest-suite': 'suite',
   'parametrization': 'parametrization',
-  'testcase': 'testcase'
+  'case': 'case'
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Common/defaults.js]: Why is this changing from testcase -> case ...?

File: testplan/web_ui/testing/src/Common/defaults.js [hunk 9]
```diff
@@ -62,7 +62,7 @@ const CATEGORY_ICONS = {
   'boost-test-suite': 'BS',
   'gtest-suite': 'GS',
   'parametrization': 'P',
-  'testcase': 'C'
+  'case': 'C'
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Common/defaults.js]: I think this should be updated to `'testcase': 'TC'` now.

File: testplan/web_ui/testing/src/Common/sampleReports.js [hunk 10]
```diff
@@ -55,7 +55,7 @@ const TESTPLAN_REPORT = {
           },
           "entries": [
             {
-              "category": "testcase",
+              "category": 'case',
```
Comment (ryan-collingham) [inline testplan/web_ui/testing/src/Common/sampleReports.js]: Why are these changed...?

File: testplan/runnable/interactive/base.py (hunk 4 above)
Comment (Pyifan) [reply testplan/runnable/interactive/base.py]: just for consistency with context
